content/5_Plat/Bin_Jump.mdx

```diff
@@ -215,9 +214,105 @@ export const problems = {
 	<Resource source="SecondThread" title="Tree Basics - Binary Lifting" url="https://codeforces.com/blog/entry/81527"> </Resource>
 </Resources>
 
+<Info>
+
+Binary jumping is more commonly referred to as "binary lifting."
+
+</Info>
+
 ### Solution
 
-<IncompleteSection />
+To solve this problem, we can use a standard binary lifting implementation where ```jmp(int x, int d)``` corresponds to the $d$th ancestor of $x$.
+
+In our ```jmp(int x, int d)``` if our final value of $x$ is $0$, then such a node does not exist and we can simply return $-1$. This is because the lowest number a node can be is $1$ (the root of the tree). 
+
+In our ```jmp(int x, int d)``` implementation, we can test if we jump in powers of two by using the $\&$ operator. If the $i$th bit on the right is toggled, then we jump. For example, a jump of $13$ would correspond to the binary number $1101$. We would jump 3 times on bits $$i$$.
+
+<Spoiler title="Illustration of the jump method">
+
+![Binary Lifting Diagram](./assets/binaryliftingjump.png)
+
+</Spoiler>
+
+To calculate the columns required for the ```up``` array, use the formula $\textrm{ceil}(\log_2{N})$ which in our case simplifies to $\textrm{ceil}(\log_2{(2\cdot 10^5)}) = 18$.
+
+
+<Info>
+
+It never hurts to add additional rows - or columns, depending on your implementation -  (as long as it's reasonable)!
+
+</Info>
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define FORE(i, a, b) for(int i = (a); i <= (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+
+int N, Q;
+
+const int MX = 2e5+5; // maximum N
+const int MS = 18; // 18 is calculated by $\log_2 2e5$
+
+int up[MS][MX]; 
+
+vector<int> adj[MX]; // adjacency list
+
+void dfs(int x) { // build jump table
+    FOR(i, 1, MS) up[i][x] = up[i-1][up[i-1][x]];
+    trav(y,adj[x]) if (y != up[0][x]) {
+        up[0][y]=x;
+        dfs(y);
+    }
+}
+
+int jmp(int x, int d) {
+    F0R(i, MS) if((d >> i) & 1) x = up[i][x];
+    return x ? x : -1; // modfication to return -1 if not found
+}
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0);
+
+    cin >> N >> Q;
+
+    FOR(i, 2, N+1) {
+        int a; cin >> a;
+        adj[a].push_back(i);
+        adj[i].push_back(a);
```

Comment:
it's sufficient to just do `adj[a].push_back(i);`.

Reply:
well, then you might as well not DFS at all